src/xtgeo/grid3d/grid_property.py

```diff
@@ -71,39 +90,98 @@ def _data_reader_factory(fformat):
     elif fformat == "bgrdecl":
         return import_bgrdecl_prop
 
-    elif fformat in ["xtg"]:
+    elif fformat == "xtg":
         return import_xtgcpprop
     else:
         raise ValueError(f"Invalid grid property file format {fformat}")
 
 
-def gridproperty_from_file(*args, **kwargs):
-    """Make a GridProperty instance directly from file import.
+def gridproperty_from_file(
+    pfile: Union[str, pathlib.Path, io.BytesIO, io.StringIO],
```

Comment:
Should we make a typing module for these unions, used in many modules? E.g. `xtgeo.typing.FileLike`